### PR TITLE
fix(ec2): two undefined tunnelInsideCidr should not throw error

### LINKED ISSUE #27772: aws-ec2: for VpnConnection, tunnelInsideCidr is required if two tunnel options
### Describe the bug

When setting up a `VpnConnection` and under `tunnelOptions` specify two `VpnTunnelOptions`, currently you are required to hard code two unique `tunnelInsideCidr`.

### Expected Behavior

That the `tunnelInsideCidr` would be automatically generated for each of the tunnels

### Current Behavior

CDK fails with following error:

```text
Error: Same undefined `tunnelInsideCidr` cannot be used for both tunnels.
``` 

### Reproduction Steps

```java
var vpc = new Vpc( /* ... */ );
vpc.addVpnConnection(
      "MyVPC",
      VpnConnectionOptions
        .builder()
        .ip(VPN_HOST)
        .staticRoutes(VPN_PEERS)
        .tunnelOptions(
          ImmutableList.of(
            VpnTunnelOption.builder().preSharedKeySecret(vpnSecret.getSecretValue()).build(),
            VpnTunnelOption.builder().preSharedKeySecret(vpnBackupSecret.getSecretValue()).build()
          )
        )
        .build()
    );
``` 

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.95.1 (build ae455d8)

### Framework Version

_No response_

### Node.js Version

v18.14.2

### OS

Linux

### Language

Java

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by msambol**
Closes #27772.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.vpn-two-tunnels.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        },
+        {
+          preSharedKeySecret: SecretValue.unsafePlainText('secretkey5678'),
+        },
+      ],
+    },
+  },
+});
+
+new integ.IntegTest(app, 'aws-cdk-redshift-elastic-ip-test', {
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.vpn-two-tunnels.ts:28]:
```suggestion
new integ.IntegTest(app, 'aws-cdk-ec2-vpn-two-tunnels-test', {
```

File: packages/aws-cdk-lib/aws-ec2/test/vpn.test.ts [hunk 2]
```diff
@@ -212,6 +212,28 @@ describe('vpn', () => {
...
+            {
+              preSharedKeySecret: SecretValue.unsafePlainText('secretkey5678'),
+            },
+          ],
+        },
+      },
+    })).not.toThrow();
+
+  });
+
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ec2/test/vpn.test.ts:236]:
```suggestion
  test('with two tunnel options and no tunnelInsideCidr', () => {
    // GIVEN
    const stack = new Stack();

    // WHEN
    new Vpc(stack, 'VpcNetwork', {
      vpnConnections: {
        VpnConnection: {
          ip: '192.0.2.1',
          tunnelOptions: [
            {
              preSharedKeySecret: SecretValue.unsafePlainText('secretkey1234'),
            },
            {
              preSharedKeySecret: SecretValue.unsafePlainText('secretkey5678'),
            },
          ],
        },
      },
    });

    // THEN
    Template.fromStack(stack).hasResourceProperties('AWS::EC2::VPNConnection', {
      CustomerGatewayId: {
        Ref: 'VpcNetworkVpnConnectionCustomerGateway8B56D9AF',
      },
      Type: 'ipsec.1',
      VpnGatewayId: {
        Ref: 'VpcNetworkVpnGateway501295FA',
      },
      StaticRoutesOnly: false,
      VpnTunnelOptionsSpecifications: [
        {
          PreSharedKey: 'secretkey1234',
        },
        {
          PreSharedKey: 'secretkey5678',
        },
      ],
    });
  });

```
Let's check also the template for better coverage.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good!
Just some adjustments on tests in my opinion.

**Review by sumupitchayan — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**